eox_nelp/course_experience/api/v1/views.py

```diff
@@ -0,0 +1,193 @@
+        SearchFilter,
+    ]
+    search_param = "filter[search]"
+
+
+class ExperienceView(BaseJsonAPIView):
+    """Class to set functionality of an ExperienceView.
+
+    Args:
+        BaseJsonAPIView: Inherited for the rest json api config.
+    """
+    def get_queryset(self, *args, **kwargs):
+        """Filter the queryset before being used.
+
+        Returns:
+            Queryset: queysyset using the super method, but filtered.
+        """
+        return super().get_queryset(*args, **kwargs).filter(author_id=self.request.user.id)
+
+    def create(self, request, *args, **kwargs):
+        """Perform processing for the request before use the base create method.
+        Args:
+            request: the request that arrives for create options.
+
+        Returns:
+            The return of ancestor create method with the request after processing.
+        """
+        if isinstance(request.data.get("course_id", {}), dict):
+            validate_course_key(request.data.get("course_id", {}).get("id", None))
+        validate_user(request.user)
+        request = self.change_author_data_2_request_user(request)
+        return super().create(request, *args, **kwargs)
+
+    def update(self, request, *args, **kwargs):
+        """Perform processing for the request before use the base update method.
+        Args:
+            request: the request that arrives for create options.
+
+        Returns:
+            The return of ancestor update method with the request after processing.
+        """
+        validate_user(request.user)
+        request = self.change_author_data_2_request_user(request)
+        return super().update(request, *args, **kwargs)
+
+    def change_author_data_2_request_user(self, request):
+        """Set the author object based in the request user.
+
+        Args:
+            request: The request to set the user.
+
+        Returns:
+            request: request with author updated.
+        """
+        if isinstance(request.data, QueryDict):
+            request.data._mutable = True  # pylint: disable=protected-access
+            request.data["author"] = f'{{"type": "User", "id": "{request.user.id}"}}'
+            request.data._mutable = False  # pylint: disable=protected-access
+        else:
+            request.data["author"] = f'{{"type": "User", "id": "{request.user.id}"}}'
+        return request
+
+
+class UnitExperienceView(ExperienceView):
+    """Class with  Experience view for units.
+
+    Args:
+        ExperienceView: Inherited to set experience views config.
+
+    """
+    lookup_field = "item_id"
+    lookup_url_kwarg = "item_id"
+    lookup_value_regex = r"block[\w\W]*"
+
+    def get_object(self):
+        """Check the item_id is a valid usage_key before the base get method.
+
+        Returns:
+            The return of ancestor create method with the request after processing.
+        """
+        item_id_string = self.kwargs.get("item_id", None)
+        validate_usage_key(item_id_string)
```

Comment:
first why do you need to validate that?, second if the item id is not valid this will raise an ValidationError that will generate a  500 error or how that is handled ?